text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1607 @@
+
+Unique nonzero id for the mapping. [deprecated]
+
+##### Field `memory_start`
+
+Address at which the binary (or DLL) is loaded into memory.
+
+##### Field `memory_limit`
+
+The limit of the address range occupied by this mapping.
+
+##### Field `file_offset`
+
+Offset in the binary that corresponds to the first mapped address.
+
+##### Field `filename`
+
+The object this entry is loaded from.  This can be a filename on
+disk for the main binary and shared libraries, or virtual
+abstractions like "[vdso]".
+
+##### Field `build_id`
+
+Index into string table
+A string that uniquely identifies a particular program version
+with high probability. E.g., for binaries generated by GNU tools,
+it could be the contents of the .note.gnu.build-id field.
+
+##### Field `build_id_kind`
+
+Index into string table
+Specifies the kind of build id. See BuildIdKind enum for more details [optional]
+
+##### Field `attributes`
+
+References to attributes in Profile.attribute_table. [optional]
+
+##### Field `has_functions`
+
+The following fields indicate the resolution of symbolic info.
+</details>
+
+#### Message `Function`
+
+Describes a function, including its human-readable name, system name,
+source file, and starting line number in the source.
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `id`
+
+Unique nonzero id for the function. [deprecated]
+
+##### Field `name`
+
+Name of the function, in human-readable form if available.
+
+##### Field `system_name`
+
+Index into string table
+Name of the function, as identified by the system.
+For instance, it can be a C++ mangled name.
+
+##### Field `filename`
+
+Index into string table
+Source file containing the function.
+
+##### Field `start_line`
+
+Index into string table
+Line number in source file.
+</details>
+
+<!-- messages -->
+
+### Example Payloads
+
+#### Simple Example
+
+Considering the following example presented in a modified folded format:
+
+```
+foo;bar;baz 100 region=us,trace_id=0x01020304010203040102030401020304,span_id=0x9999999999999999 1687841528000000
+foo;bar 200 region=us
+```
+
+It represents 2 samples:
+
+* one for stacktrace `foo;bar;baz` with value `100`, attributes `region=us`, linked to trace `0x01020304010203040102030401020304` and span `0x9999999999999999`, and timestamp `1687841528000000`
+* one for stacktrace `foo;bar` with value `200`, attributes `region=us`, no link, no timestamp
+
+The resulting profile in OTLP format would look like this (converted to YAML format for legibility):
+
+```yaml
+resource_profiles:
+  - resource:
+      attributes: null
+    schema_url: todo
+    scope_profiles:
+      - profiles:
+          - profile_id: 0102030405060708090a0b0c0d0e0f10
+            start_time_unix_nano: 1687841520000000
+            end_time_unix_nano: 1687841530000000
+            profile:
+              sample_type:
+                type: 4
+                unit: 5
+                aggregation_temporality: 1
+              attribute_table:
```

Comment:
I was a bit confused about this at first, and I think we could have better clarification around attributes.

- resource attributes is what we have for other signals already, no need to further clarify
- profile container attributes -- among others, you used `"/http/server_latency": 300` as an example, but I'm not sure what this relates to in the context of a profile.
- profile attributes table (lookup table for things within the profile)
- sample attributes -- can you give examples of sample attributes?
- mapping attributes -- ditto, what do you envision this having?
- location attributes -- same here

Reply:
* ProfileContainer is meant to be a container for a profile collected from some application or host. And so these attributes could be used to clarify how those profiles were collected, for example, we could include things that are relevant to the whole profile, e.g:
```
original_format: jfr
```
It is different from resource attributes because there can be multiple profiles per resource, so I figured it makes sense to have per profile attributes as well. 

* Sample attributes are things that could refer to a particular execution scope or request scope, e.g:
```yaml
controller: "payments"
path: "/v1/payments"
```

* Mapping attributes would be things that describe particular parts of executable, e.g. if it's JIT compiled or not. To be honest, I'm not an expert in this part, I think @florianl could explain this one better:
```yaml
jit_compiled: false
```

* Location attributes can be used to add additional information about code segments, e.g if you have a profiler that can distinguish between native stacks and interpreter stacks we could do something like this:
```yaml
language: python
native: false
```

---

By the way, none of these names are final, these are just examples. We're planning to use semantic conventions to standardize how these attributes are used after we merge this OTEP.